packages/docs/.vuepress/components/Toast.vue

```diff
@@ -0,0 +1,75 @@
+<template>
```

Comment:
Would prefer we don't invest time building Vue Components that map to our newly designed components, as this creates debt right away. We **do not** want developers to copy or use these Vue components.

Are we doing this because we don't have a playground to showcase the different variants?

Reply:
Do you have an alternate suggestion for showcasing the appropriate behavior inside the docs + providing a spec for how to handle this functionality in other frameworks?

e.g. We explicitly queue the DOM removal off of the CSS animation api. Without a functional component, I'm unsure how we demo that correctly.

Reply:
We do this primarily because there is a need to demonstrate how this functions in our documentation from an interaction perspective.

I agree that investing time in Vue components isn't great. I would much rather spend my time writing web components  _that will be used in product_ instead of Vue. Given where we are with charter talks,  and the work required to bring in WC's we just aren't there yet. This should be temporary, the sooner we can get WC's in the sooner we can move away from effectively creating prototypes.

We also arent shipping these components in any way shape or form other than in our documentation. If a developer copies this code out of the repository for use in their product, that is a sign of a different issue in my opinion.

Reply:
> Do you have an alternate suggestion for showcasing the appropriate behavior inside the docs + providing a spec for how to handle this functionality in other frameworks?

This is where UI Core comes in with web/React components - or we leverage a different playground (i.e Storybook). We're now providing another non-released component that may confuse designers and engineers, similar to the Side-bar component that's been brought up a few times.

> We do this primarily because there is a need to demonstrate how this functions in our documentation from an interaction perspective.

This is why we should prioritize Storybook as our harness for demonstrating how components operate on the page. Showing developers how something looks without providing the means to do so should be treated as a gap in our documentation. If an example is required - we shouldn't depend on a framework and rely on raw JS.

Reply:
> This is why we should prioritize Storybook as our harness for demonstrating how components operate on the page. 

Fully agree and have been a big proponent for Storybook on our end. However, this type of functionality must be defined and demonstrated by our team on the docs site as well. I acknowledge that in the future, this will be a solved issue because web components will be available. However, we continue to ship actively and need to solve this in the near/medium term.

> we shouldn't depend on a framework and rely on raw JS.

I'm OK with this as well. However, I'm not sure how this is any less confusing than shipping Vue components. It seems like we just as well do not want developers to copy or use raw JS either.

I'll also add that we _already have an example of this in our codebase_ with the tabs implementation. We don't ship it publicly, and to date, it has not been an issue. No one is leveraging our implementation over web components, because we don't ship it. It's just part of our docs package.